lib/Geo/GeoJson.cpp

```diff
@@ -353,40 +355,68 @@ Result parsePolygon(VPackSlice const& vpack, ShapeContainer& region) {
           std::string("Invalid loop in polygon: ").append(error.text()));
     }
 
+    // Note that we are using InitNested for S2Polygon below.
+    // Therefore, we are supposed to deliver all our loops in CCW
+    // convention (aka right hand rule, interiour is to the left of
+    // the polyline).
+    // Since we want to want to allow for loops, whose interiour covers
+    // more than half of the earth, we must not blindly "Normalize"
+    // the loops, as we did in earlier versions, although RFC7946
+    // says "parsers SHOULD NOT reject Polygons that do not follow
+    // the right-hand rule". Since we cannot detect if the outer loop
+    // respects the rule, we cannot reject it. However, for the other
+    // loops, we can be a bit more tolerant: If a subsequent loop is
+    // not contained in the first one (following the right hand rule),
+    // then we can invert it silently and if it is then contained
+    // in the first, we have proper nesting and leave the rest to
+    // InitNested. This is, why we proceed like this:
     S2Loop* loop = loops.back().get();
-    // normalization ensures that point orientation does not matter for Polygon
-    // type the RFC recommends this for better compatibility
-    loop->Normalize();
+    if (legacy) {
+      loop->Normalize();
+    }
 
-    // subsequent loops must be holes within first loop
+    // subsequent loops must be holes within first loop:
     if (loops.size() > 1 && !loops.front()->Contains(loop)) {
```

Comment:
does this cover the case in which there is more then one hole inside the outer loop? I mean because of checking if the first loop, which is the outer one, contains the last loop in the vector. Like if there's the outer loop followed by two holes, wouldn't it have to check if loops[1] is contained as well?

Reply:
So from my understanding, the scenario in the comment made by me above wouldn't happen, because it would be classified as a multipolygon and hence parsed in the other method?

Reply:
The first loop in the polygon is simply taken as is. All following polygons are tested for containment in the first. If they are contained, they are properly nested, if not, they are inverted and we take the only sensible interpretation. This is done for each and any loop after the first one.
S2 tests separately that no two edges of these loops intersect each other. This is true for each loop individually (so each loop is not self-intersecting), and for edges from different loops.
As a consequence it follows, that if all loops are properly contained in the first one, and no two intersect, the set of loops forms a hierarchy of properly nested loops. Yes, the third can, for example, lie within the second and so on.
This is exactly the input which `InitNested` needs.
So as far as I see, all is well.
No, such a polygon is always parsed as a single polygon and never as a multipolygon.
However, if you look beneath the surface, it turns out that polygons and multipolygons are actually processed in exactly the same way within S2, as far as I know.